Proxmoxve: Support static network configuration from cloud-init

jdoss opened the pull request:
This PR adds in support for static network configuration from cloud-init data along with tests. It follows the same methods of configuring the network in the [initramfs](https://coreos.github.io/afterburn/usage/initrd-network-cmdline/#initrd-first-boot-network-arguments) via `afterburn-network-kargs.service`. Instead of reading from injected kargs for the network config like the VMware provider, we read mount the cidata disk and read the values out of the network-config file.

travier reviewed (commented):
From a quick look, this seems reasonable, but I haven't followed the discussion and the context closely so I'll let other review and test this :)

dustymabe commented on src/providers/proxmoxve/configdrive.rs:
```diff
@@ -16,23 +16,34 @@ pub struct ProxmoxVEConfigDrive {
-            TARGET_FS,
-            3,
-        )?;
+        let device_path = Self::find_cidata_device()
+            .ok_or_else(|| anyhow::anyhow!("could not find cidata device"))?;
```
The `/dev/disk/by-label` that was used before wasn't sufficient?" Might be worth putting that in a separate commit with reasoning here.

jdoss replied:
There is a condition where it won't have a label because labeling doesn't happen until after udev runs. So we need to use blkid to find the drive based on label instead.

dustymabe replied:
Nice.. That's a great explanation of "why". Can you put that in a separate commit with a nice commit message?

dustymabe commented on src/providers/proxmoxve/cloudconfig.rs:
```diff
@@ -33,26 +34,15 @@ pub struct ProxmoxVECloudMetaData {
 #[derive(Debug, Deserialize)]
 pub struct ProxmoxVECloudUserData {
     pub hostname: String,
-    pub manage_etc_hosts: bool,
-    pub fqdn: String,
-    pub chpasswd: ProxmoxVECloudChpasswdConfig,
-    pub users: Vec<String>,
-    pub package_upgrade: bool,
```
What is it OK to delete this? Was nothing using it? 

Maybe in a separate first commit we should delete the things that are unused first?

jdoss replied:
Yes they are not used and it can cause serde errors if they are not present. Specifically package_upgrade if missing from the YAMLs it would cause afterburn to fail.

dustymabe reviewed (commented):
Thanks for working on this @jdoss. Sorry it's taken so long to get around to a review. I'm no rust programmer (only done a little learning here and there) so can't comment much on syntactical errors, but I can comment on logic and structure.

Can we run cargo fmt in a first commit and then only add properly formatted code in new commits?

Also it seems like we do a lot in the first commit. Can we delete unneeded code in a separate initial commit and also add extra context/logging in a separate initial commit?

jdoss committed:
proxmoxve: Add support for static network configuration from cloud-init.

Introduce the ability to configure static network settings which are
read from the cloud-init configuration disk. The implementation adds
the find_cidata_device function for reading the settings from the
cloud-init drive during the initrd phase. We cannot use the
/dev/disk/by-label/cidata symlink as it is not available during the
initrd phase and and it only exists after udev runs.

See https://www.man7.org/linux/man-pages/man7/dracut.bootup.7.html

jdoss committed:
proxmoxve: Add tests for static network configuration from cloud-init.

jdoss committed:
proxmoxve: Remove unneeded fields

These fields are not used and it can cause serde errors if they are not
present. Specifically package_upgrade if missing from the YAML it would
cause afterburn to fail.

jdoss committed:
proxmoxve: Add more context to log messages.

jdoss committed:
Update release notes.
